### PR TITLE
feat: accelerate stable diffusion 3 by cache and tgate and token down…

### PR DESCRIPTION
**Pull request by HaleYang**
…sample algotithm

# What does this PR do?

Fixes # (issue)

Adds # (feature)
accelerate stable diffusion 3 by cache and tgate and token downsample

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/mindspore-lab/mindone/blob/master/CONTRIBUTING.md)?
- [x] Did you make sure to update the documentation with your changes? E.g. record bug fixes or new features in `What's New`. Here are the
      [documentation guidelines](https://github.com/mindspore-lab/mindcv/wiki/%E6%96%87%E6%A1%A3%E7%BC%96%E5%86%99%E8%A7%84%E8%8C%83)
- [x] Did you build and run the code without any errors?
- [x] Did you report the running environment (NPU type/MS version) and performance in the doc? (better record it for data loading, model inference, or training tasks)
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@xxx

### CODE REVIEW

File: examples/dit_acc/stable_diffusion_3/QuickStart.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+## Quick Start
+
+### Requirement
+
+```
+mindone==0.2.0
+mindspore==2.4.0
+numpy==2.1.3
+transformers==4.38.2
+```
```
Comment (ZOULQ) [inline examples/dit_acc/stable_diffusion_3/QuickStart.md:10]: 参考其他模型，CANN版本是否需要给出

File: examples/dit_acc/stable_diffusion_3/QuickStart.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+### Usage
+
+```python
+import mindspore as ms
+from mindone.diffusers import StableDiffusion3Pipeline
+
+pipe = StableDiffusion3Pipeline.from_pretrained(
+    "path_to_checkpoint",
+    custom_pipeline="path/to/specific_pipeline_file",
```
Comment (ZOULQ) [inline examples/dit_acc/stable_diffusion_3/QuickStart.md:20]: 建议最后给出pipeline_stable_diffusion_3.py具体文件名，前面路径可以模糊

File: examples/dit_acc/stable_diffusion_3/models/transformers/transformer_sd3.py [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+            Used to indicate denoising step.
+        block_controlnet_hidden_states: (`list` of `mindspore.Tensor`):
+            A list of tensors that if specified are added to the residuals of transformer blocks.
+        joint_attention_kwargs (`dict`, *optional*):
+            A kwargs dictionary that if specified is passed along to the `AttentionProcessor` as defined under
+            `self.processor` in
+            [diffusers.models.attention_processor](https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/attention_processor.py).
+        return_dict (`bool`, *optional*, defaults to `False`):
+            Whether or not to return a [`~models.transformer_2d.Transformer2DModelOutput`] instead of a plain
+            tuple.
```
Comment (ZOULQ) [inline examples/dit_acc/stable_diffusion_3/models/transformers/transformer_sd3.py:42]: if_skip/delta_cache/delta_cache_hidden/use_cache---这几个入参没有说明

File: examples/dit_acc/README.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# DiT Acceleration
```
Comment (CaitinZhao) [inline examples/dit_acc/README.md:1]: 目录名改一下，改成 dit_infer_accleration

File: examples/dit_infer_acceleration/stable_diffusion_3/QuickStart.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
+## Quick Start
```
Comment (CaitinZhao) [inline examples/dit_infer_acceleration/stable_diffusion_3/QuickStart.md:1]: QuickStart.md -> README.md

File: examples/dit_infer_acceleration/stable_diffusion_3/QuickStart.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+Options:
+- `--ckpt`: path to Stable Diffusion 3 pretrained checkpoint
+- `--prompt`: prompt or prompt list for batch images
+- `--negative_prompt`: negative prompt, the quantity should correspond to the arg "prompt"
+- `--cache_and_prompt_gate`: apply DiTCache and PromptGate algorithms
+- `--todo`: apply ToDo algorithm
+- `--max_seq_len`: maxmium input prompt length
+- `--image_size`: output image size, default value is [1024, 1024](height, width)
+- `--save_path`: output image save path
+- `--use_graph_mode`: use graph mode to accelerate, there will be **warm up** if using graph mode
```
Comment (CaitinZhao) [inline examples/dit_infer_acceleration/stable_diffusion_3/QuickStart.md:34]: add performance data?

Comment (HaleYang) [reply examples/dit_infer_acceleration/stable_diffusion_3/QuickStart.md]: done

File: examples/dit_acc/README.md (hunk 4 above)
Comment (HaleYang) [reply examples/dit_acc/README.md]: done

File: examples/dit_infer_acceleration/stable_diffusion_3/QuickStart.md (hunk 5 above)
Comment (HaleYang) [reply examples/dit_infer_acceleration/stable_diffusion_3/QuickStart.md]: done

File: examples/dit_acc/stable_diffusion_3/QuickStart.md (hunk 1 above)
Comment (HaleYang) [reply examples/dit_acc/stable_diffusion_3/QuickStart.md]: done

File: examples/dit_acc/stable_diffusion_3/QuickStart.md (hunk 2 above)
Comment (HaleYang) [reply examples/dit_acc/stable_diffusion_3/QuickStart.md]: done

**Review by CaitinZhao — APPROVED** [review]

**Review by vigo999 — APPROVED** [review]

**Pull request merged**